File: src/main/java/liqp/filters/Append.java
package liqp.filters;

import liqp.nodes.RenderContext;

class Append extends Filter {

    /*
     * (Object) append(input, string)
     *
     * add one string to another
     */
    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        return super.asString(value) + super.asString(super.get(0, params));
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/Append.kt
package liqp.filter

import liqp.context.LContext

/**
 * {{ foo.bar | append:'john' }}
 *
 * add one string to another
 */
class Append : LFilter() {

  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    context.run {
      val content = asString(value) ?: ""
      val append: String = asString(params[0]) ?: return content
      return content + append
    }
  }
}
